spinnaker-monitoring-daemon/spinnaker-monitoring/prometheus_service.py

```diff
@@ -239,11 +239,9 @@ def __collect_instance_info(
     """
     # In practice this converts a Spinnaker Timer into either
     # <name>__count or <name>__totalTime and removes the "statistic" tag.
```

Comment:
Do we need to replace `-` with `_` in the `name` here somewhere for Prometheus?
Hyphens are not permitted per: https://prometheus.io/docs/concepts/data_model/
The HA services for echo/clouddriver/etc all have a hyphen in the name.

Reply:
Thanks for the heads up. I'll add that as a seperate PR so it can be patched into old releases if needed.

Reply:
how is that being used in the name? would that not be a tag value?

Reply:
Sorry I had a further look and the Spinnaker App name (eg: `clouddriver) is being called `service` and is used here:
https://github.com/spinnaker/spinnaker-monitoring/blob/master/spinnaker-monitoring-daemon/spinnaker-monitoring/prometheus_service.py#L145

I think we need to run a regex on the `service` value to replace hyphens or drop whatever is after the hyphen in the service name?

Reply:
Fixed here https://github.com/spinnaker/spinnaker-monitoring/pull/113